dcap/types/src/certification_data.rs

```diff
@@ -132,13 +178,41 @@ mod test {
 
     extern crate alloc;
     use alloc::vec::Vec;
+    use yare::parameterized;
 
     #[test]
-    fn zero_certification_data() {
+    fn zero_type_is_error() {
         let bytes = [0u8; MIN_CERT_DATA_SIZE];
+        assert_eq!(
+            CertificationData::try_from(bytes.as_slice()),
+            Err(Quote3Error::CertificationDataType(0))
+        );
+    }
```

Comment:
The usage of `Err` adds an extra layer of nesting that seems unnecessary to me. Consider changing it to the following:
```suggestion
        let error = CertificationData::try_from(bytes.as_slice()).unwrap_err();
        assert_eq!(error, Quote3Error::CertificationDataType(0));       
     }     
```

Reply:
separated the result -> Err logic as suggested